nuxeo-core/nuxeo-core-convert/src/main/java/org/nuxeo/ecm/core/convert/service/ConversionServiceImpl.java

```diff
@@ -219,6 +231,97 @@ public static String getCacheBasePath() {
     }
 
     @Override
+    public Blob convertBlobToPDF(Blob blob) throws IOException {
+        String mimetype = blob.getMimeType();
+        if (MimetypeRegistry.PDF_MIMETYPE.equals(mimetype)) {
+            return blob;
+        }
+        Blob result;
+        if (MediaType.TEXT_PLAIN.equals(mimetype)) {
+            result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
+        } else {
+            // Convert the blob to HTML
+            if (!MediaType.TEXT_HTML.equals(mimetype)) {
+                String filename = blob.getFilename();
+                blob = convertBlobToMimeType(blob, MediaType.TEXT_HTML);
+                blob.setFilename(filename);
+            }
+            // Replace the image URLs by absolute paths
+            Path tempDirectory = Framework.createTempDirectory("blobs");
+            DownloadService downloadService = Framework.getService(DownloadService.class);
+            blob = replaceURLsByAbsolutePaths(blob, tempDirectory, downloadService::resolveBlobFromDownloadUrl);
+            // Convert the blob to PDF
+            result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
+            org.apache.commons.io.FileUtils.deleteQuietly(tempDirectory.toFile());
```

Comment:
Put the `deleteQuietly` in a `finally` block, with the `try` just after the creation of the temp directory.